core/src/main/java/org/fao/geonet/kernel/datamanager/base/BaseMetadataCategory.java

```diff
@@ -65,7 +68,7 @@ public void apply(@Nonnull Metadata entity) {
      */
     @Override
     public boolean isCategorySet(final String mdId, final int categId) throws Exception {
-        Set<MetadataCategory> categories = getMetadataRepository().findOne(mdId).getMetadataCategories();
+        Set<MetadataCategory> categories = getMetadataUtils().findOne(mdId).getMetadataCategories();
```

Comment:
Take care using `findOne` with `String` value. I know this code has not been added new, but at least in the MetadataRepository it uses directly the `entityManager`: https://github.com/geonetwork/core-geonetwork/blob/develop/domain/src/main/java/org/fao/geonet/repository/MetadataRepositoryImpl.java#L77

And supposedly should do the same as when using the one with integer parameter (generated by JPA). 

See additional info in http://osgeo-org.1560.x6.nabble.com/Strange-problem-with-Geonetwork-in-multinode-mode-td5300933.html, should be better to convert the `String` to `Integer` when calling `findOne`. The `String` parameter is something from legacy code and should be promoted to use `Integer`

Reply:
Good suggestion, but I wouldn't change it here. This is just a refactor...